wasm/src/algorithms/poseidon/mod.rs

```diff
@@ -116,4 +119,82 @@ mod tests {
             assert_eq!(hash_8_many, native_hash_8_many);
         }
     }
+
+    #[wasm_bindgen_test]
+    fn test_wasm_literal_psd_hashes_equal_native_hashes() {
+        // Create all instances of exported hasher structs.
+        let poseidon2 = Poseidon2::new();
+        let poseidon4 = Poseidon4::new();
+        let poseidon8 = Poseidon8::new();
+
+        // Create native instances of hasher structs.
+        let native_poseidon2 = Poseidon2Native::setup("AleoPoseidon2").unwrap();
+        let native_poseidon4 = Poseidon4Native::setup("AleoPoseidon4").unwrap();
+        let native_poseidon8 = Poseidon8Native::setup("AleoPoseidon8").unwrap();
+
+        let address = Address::zero();
+
+        let literals = vec![
+            Literal::Address(address),
+            Literal::Boolean(Boolean::new(false)),
+            Literal::Field(snarkvm_console::types::Field::from_u8(2)),
+            Literal::Group(
+                snarkvm_console::types::Group::from_str(
+                    "7243206743250892049702172909169115544952822465955921992746259936160368017976group",
+                )
+                .unwrap(),
+            ),
+            Literal::Scalar(snarkvm_console::types::Scalar::from_field_lossy(&snarkvm_console::types::Field::from_u8(
+                2,
+            ))),
+        ];
+
+        let native_literals = vec![
+            LiteralNative::Address(address),
+            LiteralNative::Boolean(Boolean::new(false)),
+            LiteralNative::Field(FieldNative::from_u8(2)),
+            LiteralNative::Group(
+                GroupNative::from_str(
+                    "7243206743250892049702172909169115544952822465955921992746259936160368017976group",
+                )
+                .unwrap(),
+            ),
+            LiteralNative::Scalar(ScalarNative::from_field_lossy(&snarkvm_console::types::Field::from_u8(2))),
+        ];
+
+        for i in 0..literals.len() {
+            let literal_fields = Plaintext::Literal(literals[i].clone(), OnceCell::new()).to_fields().unwrap();
```

Comment:
Here you're using the `Plaintext` exported from SnarkVM, so it's the exact same type as `PlaintextNative` and thus this test doesn't test anything just yet about the was representation. This should just use the `to_fields` method on the exported plaintext type and you won't need to do the explicit conversion here with the macro.